scheduler/utilization_based_host_allocator.go

```diff
@@ -23,10 +23,19 @@ type TaskGroupData struct {
 	Info  model.TaskGroupInfo
 }
 
-func UtilizationBasedHostAllocator(ctx context.Context, hostAllocatorData HostAllocatorData) (int, error) {
+func UtilizationBasedHostAllocator(ctx context.Context, hostAllocatorData HostAllocatorData) (int, int, error) {
 	distro := hostAllocatorData.Distro
 	numExistingHosts := len(hostAllocatorData.ExistingHosts)
 	minimumHostsThreshold := distro.HostAllocatorSettings.MinimumHosts
+
+	// calculate approximate number of free hosts for the distro-scheduler-report
```

Comment:
this is replaced by a less approximate number if we return at the end of the function